File: Java/242. Valid Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        HashMap<Character,Integer> map = new HashMap<Character,Integer>();

        for(int i = 0; i < s.length(); i++){
            char ci = s.charAt(i);
            if(map.containsKey(ci)){
                int value = map.get(ci);
                value++;
                map.put(ci, value);
            }
            else{
                map.put(ci, 1);
            }
        }

        System.out.println(map);
        for(int j = 0; j < t.length(); j++){
            char cj = t.charAt(j);
            if(map.containsKey(cj)){
                int value = map.get(cj);
                value--;
                map.put(cj, value);
            }
            else{
                return false;
            }
        }

        for (Map.Entry m : map.entrySet()){
            if((int)m.getValue() >= 1)
                return false;
        }
        
        return true;
    }
}

File: Cpp/Data Structures I/Day 6/242. Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> map;

        for (char x: s){
            map[x]++;
        }

        for (char y: t){
            if(map[y] > 0){
                map[y]--;
            }
            else{
                return false;
            }
        }

        for(auto i : map){
            if(i.second > 0){
                return false;
            }
        }
        return true;
    }
};
